docs/03-code-internals/14-modifyclass.md

```diff
@@ -1,71 +1,138 @@
 ---
-title: Using modifyClass for objects which are initialized early in boot
-short_title: ModifyClass
+title: Using modifyClass to modify core behaviour
+short_title: modifyClass
 id: modifyclass
-
 ---
-When using modifyClass in an initializer, you may see this warning in the console:
 
-> Attempted to modify "{name}", but it was already initialized earlier in the boot process (e.g. via a lookup()). Remove that lookup, or move the modifyClass call earlier in the boot process for changes to take effect.
+For advanced themes and plugins, Discourse offers the `modifyClass` system. This allows you to extend and override functionality in many of core's javascript classes.
 
-In theme/plugin development, there are two ways this error is normally introduced:
+## When to use `modifyClass`
+
+`modifyClass` should be a last resort, when your customization cannot be made via Discourse's more stable customization APIs (e.g. plugin-api methods, plugin outets, transformers).
+
+Core's code can change at any time. And therefore, customizations made via `modifyClass` could break at any time. When using this API, you should ensure that you have controls in place to catch those issues before they reach a production site. For example, you could add automated tests to the theme/plugin, or you could use a staging site to test incoming Discourse updates against your theme/plugin.
 
-### Adding a `lookup()` caused the error
+## Basic Usage
 
-If you `lookup()` a singleton too early in the boot process, it will cause any later `modifyClass` calls to fail. In this situation, you should try to move the lookup to happen later. For example, you would change something like this:
+`api.modifyClass` can be used to modify the functions and properties of any class which is accessible via the Ember resolver. That includes Discourse's routes, controllers, services and components.
 
+`modifyClass` takes two arguments:
+
+- `resolverName` (string) - construct this by using the type (e.g. component/controller/etc.), followed by a colon, followed by the (dasherized) filename name of the class. For example: `component:d-button`, `controller:user`, `route:application`, etc.
```

Comment:
We can touch on nested components here as well.

Reply:
Added a nested example :+1: